demos/pedestrian_tracker_demo/cpp/main.cpp

```diff
@@ -175,26 +181,29 @@ int main(int argc, char **argv) {
             tracker->Process(frame, detections, cur_timestamp);
 
             presenter.drawGraphs(frame);
+            // Drawing colored "worms" (tracks).
+            frame = tracker->DrawActiveTracks(frame);
 
-            if (should_show) {
-                // Drawing colored "worms" (tracks).
-                frame = tracker->DrawActiveTracks(frame);
-
-                // Drawing all detected objects on a frame by BLUE COLOR
-                for (const auto &detection : detections) {
-                    cv::rectangle(frame, detection.rect, cv::Scalar(255, 0, 0), 3);
-                }
-
-                // Drawing tracked detections only by RED color and print ID and detection
-                // confidence level.
-                for (const auto &detection : tracker->TrackedDetections()) {
-                    cv::rectangle(frame, detection.rect, cv::Scalar(0, 0, 255), 3);
-                    std::string text = std::to_string(detection.object_id) +
-                        " conf: " + std::to_string(detection.confidence);
-                    cv::putText(frame, text, detection.rect.tl(), cv::FONT_HERSHEY_COMPLEX,
-                                1.0, cv::Scalar(0, 0, 255), 3);
-                }
+            // Drawing all detected objects on a frame by BLUE COLOR
+            for (const auto &detection : detections) {
+                cv::rectangle(frame, detection.rect, cv::Scalar(255, 0, 0), 3);
+            }
 
+            // Drawing tracked detections only by RED color and print ID and detection
+            // confidence level.
+            for (const auto &detection : tracker->TrackedDetections()) {
+                cv::rectangle(frame, detection.rect, cv::Scalar(0, 0, 255), 3);
+                std::string text = std::to_string(detection.object_id) +
+                    " conf: " + std::to_string(detection.confidence);
+                cv::putText(frame, text, detection.rect.tl(), cv::FONT_HERSHEY_COMPLEX,
+                            1.0, cv::Scalar(0, 0, 255), 3);
+            }
+
+            framesProcessed++;
+            if (videoWriter.isOpened() && (FLAGS_output_limit == 0 || framesProcessed <= FLAGS_output_limit)) {
+                videoWriter.write(frame);
+            }
+            if (should_show) {
                 cv::resize(frame, frame, cv::Size(), 0.5, 0.5);
```

Comment:
The shown image is two times smaller than one saved. But the stranger is that this resize even exists. Will you remove it?

Reply:
Removed.